pkg/app/app.go

```diff
@@ -125,6 +126,29 @@ func defaultInputDirectory() configuration.DefaultValueFunction {
 	return callback
 }
 
+func defaultTempDirectory(engine workflow.Engine, logger *zerolog.Logger) configuration.DefaultValueFunction {
+	callback := func(existingValue interface{}) interface{} {
+		if existingValue != nil {
+			return existingValue
+		}
+		ri := engine.GetRuntimeInfo()
+		if ri == nil {
+			return false
+		}
+
+		version := ri.GetVersion()
```

Comment:
Suggestion: let's just check that the version isn't empty and use some default value if it is.